src/tools/clippy/clippy_lints/src/lib.rs

```diff
@@ -43,6 +43,7 @@ extern crate rustc_session;
 extern crate rustc_span;
 extern crate rustc_target;
 extern crate rustc_trait_selection;
+extern crate thin_vec;
```

Comment:
(un)educated guess here: You might have to specify the `thin-vec` dependency in [`rustc_workspace_hack`](https://github.com/rust-lang/rust/tree/master/src/tools/rustc-workspace-hack), but I'm not 100% sure that this will help. Using `extern crate` here is definitely something that we don't want to do here.

Reply:
I tried fiddling with `rustc_workspace_hack` but it didn't help. AIUI that's all about making sure that a consistent feature set is used for each crate, but that's not relevant here because rustc isn't using any `thin-vec` features.

> Is `thin_vec` shipped with `rustc-dev`?

I don't know what `rustc-dev` is. Some googling turned up https://rust-lang.github.io/rustup/concepts/components.html, which certainly sounds relevant. I found some mentions of `rustc-dev` in the rust repo, but I couldn't work out what crates are part of it, and how to add `thin_vec` to it. Any suggestions?

Reply:
It seems the `extern crate thin_vec` is needed because `ThinVec` is part of the interface for the AST, which clippy uses. From @cuviper: "If the crate is in any shared rustc API, then they have to use the same compilation of that crate."

rustdoc already has some `extern crate` declarations like this, and it gets another one in this PR for `thin_vec`. Seems like `clippy` is in the same situation.